pkg/oc/cli/policy/cani/cani.go

```diff
@@ -199,40 +201,73 @@ func (o *CanIOptions) Run() (bool, error) {
 		fmt.Fprintln(o.Out)
 	}
 
-	return response.Allowed, nil
+	// honor existing exit code convention
+	if !response.Allowed && o.Quiet {
+		os.Exit(2)
+	}
+
+	return nil
 }
 
 func (o *CanIOptions) listAllPermissions() error {
-	var rulesReviewResult runtime.Object
+	var rulesReviewObj runtime.Object
+	var rulesReviewResult []authorizationv1.PolicyRule
 
 	if len(o.User) == 0 && len(o.Groups) == 0 {
-		rulesReview := &authorizationapi.SelfSubjectRulesReview{}
+		rulesReview := &authorizationv1.SelfSubjectRulesReview{}
 		if len(o.Scopes) > 0 {
 			rulesReview.Spec.Scopes = o.Scopes
 		}
 
-		whatCanIDo, err := o.SelfRulesReviewClient.SelfSubjectRulesReviews(o.Namespace).Create(rulesReview)
+		whatCanIDo, err := o.AuthClient.SelfSubjectRulesReviews(o.Namespace).Create(rulesReview)
 		if err != nil {
 			return err
 		}
 
-		rulesReviewResult = whatCanIDo
+		rulesReviewResult = whatCanIDo.Status.Rules
+		rulesReviewObj = whatCanIDo
 	} else {
-		rulesReview := &authorizationapi.SubjectRulesReview{
-			Spec: authorizationapi.SubjectRulesReviewSpec{
+		rulesReview := &authorizationv1.SubjectRulesReview{
+			Spec: authorizationv1.SubjectRulesReviewSpec{
 				User:   o.User,
 				Groups: o.Groups,
 				Scopes: o.Scopes,
 			},
 		}
 
-		whatCanYouDo, err := o.RulesReviewClient.SubjectRulesReviews(o.Namespace).Create(rulesReview)
+		whatCanYouDo, err := o.AuthClient.SubjectRulesReviews(o.Namespace).Create(rulesReview)
 		if err != nil {
 			return err
 		}
 
-		rulesReviewResult = whatCanYouDo
+		rulesReviewResult = whatCanYouDo.Status.Rules
+		rulesReviewObj = whatCanYouDo
 	}
 
-	return o.Printer.PrintObj(rulesReviewResult, o.Out)
+	successOutput := bytes.NewBuffer([]byte{})
+	w := tabwriter.NewWriter(successOutput, tabWriterMinWidth, tabWriterWidth, tabWriterPadding, tabWriterPadChar, tabWriterFlags)
+	fmt.Fprintln(w)
+
+	// print columns
+	if !o.NoHeaders {
+		fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t\n", "VERBS", "NON-RESOURCE URLS", "RESOURCE NAMES", "API GROUPS", "RESOURCES")
+	}
+
+	for _, rule := range rulesReviewResult {
+		fmt.Fprintf(w, "%v\t%v\t%v\t%v\t%v\n",
+			rule.Verbs,
+			rule.NonResourceURLsSlice,
+			rule.ResourceNames,
+			rule.APIGroups,
+			rule.Resources,
+		)
+	}
+
+	w.Flush()
+
+	p, err := o.ToPrinter(successOutput.String())
```

Comment:
@soltysh printing `--list` output here, instead of using the humanreadable printHandler (like we were doing before), cuts this dependency on the internalversion printers, and makes use of the "success" output of the new printing stack